src/services/completions.ts

```diff
@@ -1036,11 +1059,18 @@ namespace ts.Completions {
                     symbolToOriginInfoMap[getSymbolId(firstAccessibleSymbol)] =
                         !moduleSymbol || !isExternalModuleSymbol(moduleSymbol) ? { kind: SymbolOriginInfoKind.SymbolMemberNoExport } : { kind: SymbolOriginInfoKind.SymbolMemberExport, moduleSymbol, isDefaultExport: false };
                 }
-                else if (preferences.includeCompletionsWithInsertText) {
-                    symbols.push(symbol);
+                else if(preferences.includeCompletionsWithInsertText) {
+                    addPromiseSymbolOriginInfo(symbol);
                 }
             }
             else {
+                addPromiseSymbolOriginInfo(symbol);
+            }
+
+            function addPromiseSymbolOriginInfo (symbol: Symbol) {
+                if (insertAwait && preferences.includeCompletionsWithInsertText && !symbolToOriginInfoMap[getSymbolId(symbol)]) {

```

Comment:
I don't think you want to push symbol to completions at all if `preferences.includeCompletionsWithInsertText` is false. Your tests don't have anything that tests this with`preferences.includeCompletionsWithInsertText`  as `false`. Please add test for with that preference as false as well. Thanks